src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs

```diff
@@ -108,6 +109,19 @@ public static class AspireAzureCosmosDBExtensions
         if (serviceKey is null)
         {
             builder.Services.AddSingleton(_ => ConfigureDb());
+
+            var csBuilder = new DbConnectionStringBuilder();
+            csBuilder.ConnectionString = settings.ConnectionString;
+
+            if (csBuilder.TryGetValue("Database", out var databaseName))
+            {
+                builder.Services.AddSingleton<Database>(sp =>
```

Comment:
This feels really weird to conditionally add a DI service based on the format used in the connection string.

Reply:
The more I think about this, the more I don't think we should be doing this in the Aspire component. If someone wants to make a separate NuGet package to do these kinds of convenience operations, that would be OK, but Aspire is more geared towards "cloud" concerns like telemetry, config, health checks, etc. Its goal isn't trying to make the underlying libraries easier to use. It registers the root concept in DI and then gets out of the user's way and let's them decide how to structure their app.

Reply:
I think this is fair. Ideally Cosmos libraries would do a lot of this for us.